video/decode/videotoolbox.c

```diff
@@ -24,8 +24,28 @@
 #include "common/msg.h"
 #include "video/mp_image.h"
 #include "video/decode/lavc.h"
+#include "video/mp_image_pool.h"
 #include "config.h"
 
+struct priv {
+    struct mp_image_pool *sw_pool;
+};
+
+static int probe_copy(struct lavc_ctx *ctx, struct vd_lavc_hwdec *hwdec,
+                 const char *codec)
+{
+    switch (mp_codec_to_av_codec_id(codec)) {
+    case AV_CODEC_ID_H264:
+    case AV_CODEC_ID_H263:
+    case AV_CODEC_ID_MPEG1VIDEO:
+    case AV_CODEC_ID_MPEG2VIDEO:
+    case AV_CODEC_ID_MPEG4:
+        break;
+    default:
+        return HWDEC_ERR_NO_CODEC;
+    }
+    return 0;
+}
```

Comment:
Should be shared with the direct probe function. (Or rather, you could make the direct probe function call this function.)